src/lib.rs

```diff
@@ -632,7 +637,24 @@ mod tests {
 
     use std::collections::BTreeMap;
 
-    use rand::{self, random};
+    use rand::{self, random, Rng};
+
+    #[test]
+    fn test_interpolate() {
+        let mut rng = rand::thread_rng();
+        for deg in 0..5 {
+            println!("deg = {}", deg);
+            let comm = Poly::random(deg, &mut rng).commitment();
+            let mut values = Vec::new();
+            let mut x = 0;
+            for _ in 0..(deg + 1) {
```

Comment:
I'm a horrible person... but `0..=deg` ?

Reply:
No, you're right. :smile: 
I amended the commit:

```diff
-            for _ in 0..(deg + 1) {
+            for _ in 0..=deg {
```